src/Lykke.Sdk/LykkeApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using Common.Log;
+using JetBrains.Annotations;
+using Lykke.Common.ApiLibrary.Middleware;
+using Lykke.MonitoringServiceApiCaller;
+using Lykke.Sdk.Settings;
+using Lykke.SettingsReader;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Lykke.Sdk
+{
+    [PublicAPI]
+    public static class LykkeApplicationBuilderExtensions
+    {
+        public static void UseLykkeConfiguration(this IApplicationBuilder app, Action<LykkeAppOptions> optionBuilder)
+        {
+            var log = app.ApplicationServices.GetService<ILog>();
+
+            try
+            {
+                var options = new LykkeAppOptions();
+                optionBuilder(options);
+
+                if (string.IsNullOrWhiteSpace(options.AppName))
+                    throw new ApplicationException("Application name is required.");
+
+                if (string.IsNullOrWhiteSpace(options.ApiVersion))
+                    throw new ApplicationException("ApiVersion is required.");
+
+                var appLifetime = app.ApplicationServices.GetService<IApplicationLifetime>();
+                var configurationRoot = app.ApplicationServices.GetService<IConfigurationRoot>();
```

Comment:
I think its better to provide IConfigurationRoot via method parameter than using container, cause some apps might not register Configuration in there

Reply:
validation added